torch/distributed/rpc/api.py

```diff
@@ -238,9 +241,31 @@ def _all_gather(obj, timeout=UNSET_RPC_TIMEOUT):
                 f"after {timeout:.2f} seconds. The first exception is {errors[0][1]}"
             )
 
+    # Clean up for the states using the sequence_id
+    with _all_gather_dict_lock:
+        states = _all_gather_sequence_id_to_states.pop(sequence_id)
     return states.gathered_objects
 
 
+@_require_initialized
+def _barrier(worker_names):
```

Comment:
Should we make it a public API?

Reply:
I had discussed this with Shen a few weeks ago as the PR was originally for public API. But since we are unsure of the people who will use it and also how the API may change when introducing Elastic RPC, decided to keep it internal for now.